id: System.ComponentModel.DataAnnotations.ValidationException
href: System.ComponentModel.DataAnnotations.ValidationException.yaml
name:
  CSharp: ValidationException
qualifiedName:
  CSharp: System.ComponentModel.DataAnnotations.ValidationException
type: Class
source:
  remote: &o0
    branch: master
    repo: https://github.com/dotnet/corefx.git
  path: src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationException.cs
  startLine: 8
summary: Exception used for validation using [System.ComponentModel.DataAnnotations.ValidationException.ValidationAttribute](System.ComponentModel.DataAnnotations.ValidationException.yaml).
syntax:
  content:
    CSharp: public class ValidationException
inheritance:
- id: Exception
  name: Exception
  isExternal: true
items:
- id: System.ComponentModel.DataAnnotations.ValidationException.#ctor(System.ComponentModel.DataAnnotations.ValidationResult,System.ComponentModel.DataAnnotations.ValidationAttribute,System.Object)
  href: System.ComponentModel.DataAnnotations.ValidationException.yaml
  name:
    CSharp: ValidationException(System.ComponentModel.DataAnnotations.ValidationResult,System.ComponentModel.DataAnnotations.ValidationAttribute,System.Object)
  qualifiedName:
    CSharp: System.ComponentModel.DataAnnotations.ValidationException.ValidationException(System.ComponentModel.DataAnnotations.ValidationResult,System.ComponentModel.DataAnnotations.ValidationAttribute,System.Object)
  type: Constructor
  source:
    remote: *o0
    path: src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationException.cs
    startLine: 18
  summary: Constructor that accepts a structured [System.ComponentModel.DataAnnotations.ValidationException.ValidationResult](System.ComponentModel.DataAnnotations.ValidationException.yaml) describing the problem.
  syntax:
    content:
      CSharp: 'public ValidationException(ValidationResult validationResult, ValidationAttribute validatingAttribute, object value): this (validationResult.ErrorMessage, validatingAttribute, value)'
    parameters:
    - id: validationResult
      type:
        id: System.ComponentModel.DataAnnotations.ValidationResult
        name: ValidationResult
        href: System.ComponentModel.DataAnnotations.ValidationResult.yaml
      description: The value describing the validation error
    - id: validatingAttribute
      type:
        id: System.ComponentModel.DataAnnotations.ValidationAttribute
        name: ValidationAttribute
        href: System.ComponentModel.DataAnnotations.ValidationAttribute.yaml
      description: The attribute that triggered this exception
    - id: value
      type:
        id: System.Object
        name: System.Object
        isExternal: true
      description: The value that caused the validating attribute to trigger the exception
    return:
      id: Void
- id: System.ComponentModel.DataAnnotations.ValidationException.#ctor(System.String,System.ComponentModel.DataAnnotations.ValidationAttribute,System.Object)
  href: System.ComponentModel.DataAnnotations.ValidationException.yaml
  name:
    CSharp: ValidationException(System.String,System.ComponentModel.DataAnnotations.ValidationAttribute,System.Object)
  qualifiedName:
    CSharp: System.ComponentModel.DataAnnotations.ValidationException.ValidationException(System.String,System.ComponentModel.DataAnnotations.ValidationAttribute,System.Object)
  type: Constructor
  source:
    remote: *o0
    path: src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationException.cs
    startLine: 31
  summary: Constructor that accepts an error message, the failing attribute, and the invalid value.
  syntax:
    content:
      CSharp: 'public ValidationException(string errorMessage, ValidationAttribute validatingAttribute, object value): base (errorMessage)'
    parameters:
    - id: errorMessage
      type:
        id: System.String
        name: System.String
        isExternal: true
      description: The localized error message
    - id: validatingAttribute
      type:
        id: System.ComponentModel.DataAnnotations.ValidationAttribute
        name: ValidationAttribute
        href: System.ComponentModel.DataAnnotations.ValidationAttribute.yaml
      description: The attribute that triggered this exception
    - id: value
      type:
        id: System.Object
        name: System.Object
        isExternal: true
      description: The value that caused the validating attribute to trigger the exception
    return:
      id: Void
- id: System.ComponentModel.DataAnnotations.ValidationException.#ctor
  href: System.ComponentModel.DataAnnotations.ValidationException.yaml
  name:
    CSharp: ValidationException
  qualifiedName:
    CSharp: System.ComponentModel.DataAnnotations.ValidationException.ValidationException
  type: Constructor
  source:
    remote: *o0
    path: src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationException.cs
    startLine: 42
  summary: Default constructor.
  syntax:
    content:
      CSharp: 'public ValidationException(): base ()'
    parameters: []
    return:
      id: Void
- id: System.ComponentModel.DataAnnotations.ValidationException.#ctor(System.String)
  href: System.ComponentModel.DataAnnotations.ValidationException.yaml
  name:
    CSharp: ValidationException(System.String)
  qualifiedName:
    CSharp: System.ComponentModel.DataAnnotations.ValidationException.ValidationException(System.String)
  type: Constructor
  source:
    remote: *o0
    path: src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationException.cs
    startLine: 52
  summary: Constructor that accepts only a localized message
  syntax:
    content:
      CSharp: 'public ValidationException(string message): base (message)'
    parameters:
    - id: message
      type:
        id: System.String
        name: System.String
        isExternal: true
      description: The localized message
    return:
      id: Void
- id: System.ComponentModel.DataAnnotations.ValidationException.#ctor(System.String,Exception)
  href: System.ComponentModel.DataAnnotations.ValidationException.yaml
  name:
    CSharp: ValidationException(System.String,Exception)
  qualifiedName:
    CSharp: System.ComponentModel.DataAnnotations.ValidationException.ValidationException(System.String,Exception)
  type: Constructor
  source:
    remote: *o0
    path: src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationException.cs
    startLine: 63
  summary: Constructor that accepts a localized message and an inner exception
  syntax:
    content:
      CSharp: 'public ValidationException(string message, Exception innerException): base (message, innerException)'
    parameters:
    - id: message
      type:
        id: System.String
        name: System.String
        isExternal: true
      description: The localized error message
    - id: innerException
      type:
        id: Exception
        name: Exception
        isExternal: true
      description: inner exception
    return:
      id: Void
- id: System.ComponentModel.DataAnnotations.ValidationException.ValidationAttribute
  href: System.ComponentModel.DataAnnotations.ValidationException.yaml
  name:
    CSharp: ValidationAttribute
  qualifiedName:
    CSharp: System.ComponentModel.DataAnnotations.ValidationException.ValidationAttribute
  type: Property
  source:
    remote: *o0
    path: src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationException.cs
    startLine: 71
  summary: Gets the ValidationAttribute instance that triggered this exception.
  syntax:
    content:
      CSharp: public ValidationAttribute ValidationAttribute { get; private set; }
    parameters:
    - id: ValidationAttribute
      type:
        id: System.ComponentModel.DataAnnotations.ValidationAttribute
        name: ValidationAttribute
        href: System.ComponentModel.DataAnnotations.ValidationAttribute.yaml
- id: System.ComponentModel.DataAnnotations.ValidationException.ValidationResult
  href: System.ComponentModel.DataAnnotations.ValidationException.yaml
  name:
    CSharp: ValidationResult
  qualifiedName:
    CSharp: System.ComponentModel.DataAnnotations.ValidationException.ValidationResult
  type: Property
  source:
    remote: *o0
    path: src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationException.cs
    startLine: 79
  summary: Gets the [System.ComponentModel.DataAnnotations.ValidationException.ValidationResult](System.ComponentModel.DataAnnotations.ValidationException.yaml) instance that describes the validation error.
  syntax:
    content:
      CSharp: public ValidationResult ValidationResult { get; }
    parameters:
    - id: ValidationResult
      type:
        id: System.ComponentModel.DataAnnotations.ValidationResult
        name: ValidationResult
        href: System.ComponentModel.DataAnnotations.ValidationResult.yaml
- id: System.ComponentModel.DataAnnotations.ValidationException.Value
  href: System.ComponentModel.DataAnnotations.ValidationException.yaml
  name:
    CSharp: Value
  qualifiedName:
    CSharp: System.ComponentModel.DataAnnotations.ValidationException.Value
  type: Property
  source:
    remote: *o0
    path: src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationException.cs
    startLine: 94
  summary: Gets the value that caused the validating attribute to trigger the exception
  syntax:
    content:
      CSharp: public object Value { get; private set; }
    parameters:
    - id: Value
      type:
        id: System.Object
        name: System.Object
        isExternal: true
